src/array_api_extra/_funcs.py

```diff
@@ -1,10 +1,22 @@
+from __future__ import annotations
```

Comment:
Needed due to `__getitem__(self, idx) -> at:`

Reply:
I think you can remove it if you change that to `__getitem__(self, idx) -> "at":`.

When we drop Python 3.10 we can then change it to `-> Self`.